private/rules/coursier.bzl

```diff
@@ -1364,7 +1364,7 @@ pinned_coursier_fetch = repository_rule(
             doc = "Instructions to re-pin the repository if required. Many people have wrapper scripts for keeping dependencies up to date, and would like to point users to that instead of the default.",
         ),
         "excluded_artifacts": attr.string_list(default = []),  # only used for hash generation
-        "_workspace_label": attr.label(default = Label("@//does/not:exist")),
+        "_workspace_label": attr.label(default = Label("@@//does/not:exist")),
```

Comment:
We still need to support Bazel 5 and 6, so this will cause a regression there. I think we need to do something smart to figure out whether or not `bzlmod` is being used (not just checking the version number), and then Do The Right Thing

Reply:
The `maven_bom.bzl` file has a utility method to determine whether we're using `bzlmod` or not. We might want to lift that out into another file and depend on that.

Reply:
@shs96c will you be dropping support for Bazel 6 once 8 is out?